apps/api/src/app/workflows-v2/usecases/get-workflow/get-workflow.usecase.ts

```diff
@@ -34,6 +34,7 @@ export class GetWorkflowUseCase {
       GetPreferencesCommand.create({
         environmentId: command.user.environmentId,
         organizationId: command.user.organizationId,
+        templateId: workflowEntity._id,
```

Comment:
We should provide `templateId` when retrieving specific workflow, otherwise it would return any preference from the org and environment I guess.

Reply:
Good catch!